### PR TITLE
(#122) Trial Description General Page Components Test 

### LINKED ISSUE #122: Migration of Trial Description general page components test
### Acceptance Criteria

```gherkin


    Scenario: Trial description page displays general page components
        Given user is navigating to "/about-cancer/treatment/clinical-trials/search/r?loc=0&rl=2"
        Then page title is "Clinical Trials Search Results"
        And trial info displayes "Results 1-10 of \d+ for your search for: \"all trials\""
        And "Show Search Criteria" button does not exist
        And "Start Over" link has a href "/about-cancer/treatment/clinical-trials/search/advanced"
        And "Modify Search Criteria" button does not exist
        When user clicks on 1 result item link
        Then user is redirected to "/about-cancer/treatment/clinical-trials/search/v" with query parameters that match "id=NCI-\d+-\d+&loc=0&rl=2"
        And "< Back to search results" link is displayed
        And "This clinical trial matches: \"all trials\"" text is displayed
        And "Start Over" link has a href "/about-cancer/treatment/clinical-trials/search/advanced"
        And trial details accordion is displayed
        And "Description" section is expanded
        And "Open all" button is displayed
        And "Close all" button is displayed
        And option to "Print" trial is displayed
        And option to "Email" trial is displayed
        And there are 2 delighters present
        And the following delighters are displayed
            | delighter    | href                                                       | title                              | text                                                                  |
            | cts-livehelp | /contact                                                   | Have a question?We're here to help | Chat with us: LiveHelpCall us: 1-800-4-CANCER(1-800-422-6237)         |
            | cts-which    | /about-cancer/treatment/clinical-trials/search/trial-guide | Which trials are right for you?    | Use the checklist in our guide to gather the information you’ll need. |
```
```gherkin
    Scenario: Trial description page displays general page components on mobile
        Given screen breakpoint is set to "mobile"
        Given user is navigating to "/about-cancer/treatment/clinical-trials/search/r?loc=0&q=breast%20cancer&rl=2"
        Then page title is "Clinical Trials Search Results"
        When user clicks on 1 result item link
        Then user is redirected to "/about-cancer/treatment/clinical-trials/search/v" with query parameters that match "id=NCI-\d+-\d+&loc=0&q=breast%20cancer&rl=2"
        And "< Back to search results" link is displayed
        And "This clinical trial matches:" text is displayed
        And "Show Search Criteria" button is displayed
        When user clicks on "Show Search Criteria" button
        Then "Hide Search Criteria" button is displayed
        And search criteria table displayes the following
            | Category         | Your Selection |
            | Keywords/Phrases | breast cancer  |
        And "Start Over" link has a href "/about-cancer/treatment/clinical-trials/search/advanced"
        And trial details accordion is displayed
        And "Description" section is expanded
        And "Open all" button is displayed
        And "Close all" button is displayed
        When user clicks on "Open all" button
        Then all trial details sections are expanded
        When user clicks on "Close all" button
        Then all trial details sections are collapsed
        And option to "Print" trial does not exist
        And option to "Email" trial is displayed
        And there are 2 delighters present
        And the following delighters are displayed
            | delighter    | href                                                       | title                              | text                                                                  |
            | cts-livehelp | /contact                                                   | Have a question?We're here to help | Chat with us: LiveHelpCall us: 1-800-4-CANCER(1-800-422-6237)         |
            | cts-which    | /about-cancer/treatment/clinical-trials/search/trial-guide | Which trials are right for you?    | Use the checklist in our guide to gather the information you’ll need. |

```

### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* Web Page: *Here goes a URL to an existing web page*

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Short Task Description - Issue #9999
- [ ] Short Task Description - Issue #9999

### Technical Notes
- More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by alinai**
Added feature file.
Added more tests.
Closes #122

### CODE REVIEW

**Comment by alinai** [comment]
@KateMashkinaNIH , please review.
image

**Comment by alinai** [comment]
image

File: cypress/integration/CTS/TrialDescriptionGeneralPageComponentsTest/index.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+And('{string} button does not exist', (showSearchCriteriaButton) => {
+    cy.get('button').contains(showSearchCriteriaButton).should('not.exist');
+});
+
+And('{string} link has a href {string}', (linkText, linkHref) => {
+    cy.get(`a[href="${linkHref}"]`).should('have.text', linkText);
+});
+
+And('{string} button does not exist', (modifySearchCriteriaButton) => {
+    cy.get('button').contains(modifySearchCriteriaButton).should('not.exist');
```
Comment (kate-mashkina) [inline cypress/integration/CTS/TrialDescriptionGeneralPageComponentsTest/index.js:18]: ths step is duplicate of step at line 9

File: cypress/integration/CTS/TrialDescriptionGeneralPageComponentsTest/index.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+And('{string} text is displayed', (infoText) => {
+    cy.get('.trial-description-page__header').find('strong').first().should('have.text', infoText).and('be.visible');
+});
+
+And('trial details accordion is displayed', () => {
+    cy.get('.trial-description-page__content  .cts-accordion').should('be.visible');
+});
+
+And('{string} section is expanded', (descriptionSection) => {
+    cy.get('div[class="cts-accordion  "] button.cts-accordion__button').contains(descriptionSection).parent()
```
Comment (kate-mashkina) [inline cypress/integration/CTS/TrialDescriptionGeneralPageComponentsTest/index.js:44]: the locator with just ('button.cts-accordion__button').contains... should suffice

File: cypress/integration/CTS/TrialDescriptionGeneralPageComponentsTest/index.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+And('search criteria table displayes the following', (dataTable) => {
+    for (const { Category, Selection } of dataTable.hashes()) {
+        cy.get('tbody tr th').should('have.text', Category);
+        cy.get('tbody tr td').should('have.text', Selection);
+    }
+});
+
+Then('all trial details sections are expanded', () => {
+    cy.get('div[class="cts-accordion  "] button.cts-accordion__button').should('have.attr', 'aria-expanded', 'true');
```
Comment (kate-mashkina) [inline cypress/integration/CTS/TrialDescriptionGeneralPageComponentsTest/index.js:81]: same here, just make locator shorter pls ( I dont like the two spaces inside quotes)

File: cypress/integration/CTS/TrialDescriptionGeneralPageComponentsTest/index.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+});
+
+Then('all trial details sections are collapsed', () => {
+    cy.get('div[class="cts-accordion  "] button.cts-accordion__button').should('have.attr', 'aria-expanded', 'false');
```
Comment (kate-mashkina) [inline cypress/integration/CTS/TrialDescriptionGeneralPageComponentsTest/index.js:85]: same locator

**Review by kate-mashkina — CHANGES_REQUESTED** [review]

File: cypress/integration/CTS/TrialDescriptionGeneralPageComponentsTest/index.js (hunk 1 above)
Comment (alinai) [reply cypress/integration/CTS/TrialDescriptionGeneralPageComponentsTest/index.js:18]: done

File: cypress/integration/CTS/TrialDescriptionGeneralPageComponentsTest/index.js (hunk 2 above)
Comment (alinai) [reply cypress/integration/CTS/TrialDescriptionGeneralPageComponentsTest/index.js:44]: done

File: cypress/integration/CTS/TrialDescriptionGeneralPageComponentsTest/index.js (hunk 3 above)
Comment (alinai) [reply cypress/integration/CTS/TrialDescriptionGeneralPageComponentsTest/index.js:81]: done

File: cypress/integration/CTS/TrialDescriptionGeneralPageComponentsTest/index.js (hunk 4 above)
Comment (alinai) [reply cypress/integration/CTS/TrialDescriptionGeneralPageComponentsTest/index.js:85]: done

**Comment by alinai** [comment]
@KateMashkinaNIH , I have updated per your review, please review again.

Test Result after updating per review,

image

**Review by kate-mashkina — APPROVED** [review]

**Pull request merged**